src/workflows/deploy.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+GIT_BRANCH="${1:-""}"
+
+fail_if_unset () {
+  VAR_NAME="$1"
+  if [[ -z "${!VAR_NAME:-""}" ]]; then
+    printf '%s not set\n' "$VAR_NAME" >&2
+    exit 1
+  fi
+}
+gwei_to_wei () {
+  GWEI="$1"
+  printf "%s000000000" "${GWEI}"
+}
+
+fail_if_unset "ETHERSCAN_API_KEY"
+fail_if_unset "INFURA_KEY"
+fail_if_unset "PK"
+
+GAS_PRICE_GWEI_MAINNET="$( \
+  curl --silent "https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=$ETHERSCAN_API_KEY" \
+    | jq -e --raw-output .result.FastGasPrice \
+)"
+if ! [[ "$GAS_PRICE_GWEI_MAINNET" =~ ^[1-9][0-9]{1,2} ]]; then
+  echo "Invalid mainnet gas price $GAS_PRICE_GWEI_MAINNET" >&2
+  exit 1
+fi
+
+yarn deploy --network rinkeby --gasprice "$(gwei_to_wei 1)"
+yarn deploy --network xdai --gasprice "$(gwei_to_wei 1)"
+yarn deploy --network mainnet --gasprice "$(gwei_to_wei "$GAS_PRICE_GWEI_MAINNET")"
+
+# wait for Etherscan to register the new contracts on the blockchain
+sleep 60
```

Comment:
nit: This script is very awesome, but that "hacky" line caught my attention 😉 

Maybe you could use the etherscan API to continuously query until this address returns a value (with a timeout of 60s or so).

Reply:
Mmh, I agree with you but thinking about it I prefer to keep this little hack.

The check you suggest should be done for
1. all contracts that have been deployed (so I'd need to make a dedicated TS script doing it, in order to keep it maintainable)
2. and both (all three) networks: rinkeby, mainnet (and xdai).

It seems a lot of work for relatively little benefit.

In my experiment the actual time needed is more in the order of the 5 seconds, so I'm being a bit cautious here.

Reply:
I'm fine with this solution until we run into issues.